src/jennifer/model/column_mapping.cr

```diff
@@ -0,0 +1,44 @@
+module Jennifer::Model
+  module ColumnMapping
+    #
+    # This macro defines a constant `COLUMN_MAPPINGS` that captures
+    # all mappings from field names to column names that were defined in the
+    # given property specification.
+    # Those mappings may be specified by specifying the mapping attribute
+    # `column` in combination with the targetted column name.
+    # E.g. a field mapping like `a: { type: Int64, column: :b}`
+    # results in a field `a` that is mapped to the column `b` in the database.
+    # The mapping constant is accessible in other macros and resolves field
+    # names to column names. In order to generalize access, it also stores
+    # fields that do not have a custom mapping - in this case the map
+    # returns the same name as the field (e.g. for above example
+    # `COLUMN_MAPPINGS["a"] == "b"`, but for another - unmapped column - `c`
+    # it holds that `COLUMN_MAPPINGS["c"] == "c"`).
+    #
+    # @param properties All property definitions of the mapped type
+    #
+    macro _register_column_mappings(properties)
+      {%
+        column_mappings = properties.keys.reduce({} of StringLiteral => StringLiteral) do |h, k|
+          v = properties[k]
+          h[k] =
+            # check if the mapping is specified as hash

```

Comment:
- remove comments inside macro body (the logic is obvious and also they are rendered to a resulting class definition which makes debugging harder)
- rename variables to something more readable: h -> mapping, k -> key (or `field`)